Source/Scene/Expression.js

```diff
@@ -226,6 +281,33 @@ define([
         return this._value;
     };
 
+    Node.prototype._evaluateVariableString = function(feature) {
+        var exp = this._value;
+        var result = "";
+        var i = exp.indexOf('${');
```

Comment:
There is no need to iterate the string manually at all.  This can be done with a couple of regexes and string.replace. My comment about escaped `$` also applies here.